tests/db_functions/math/test_round.py

```diff
@@ -51,3 +53,74 @@ def test_transform(self):
             self.assertEqual(obj.n1, Decimal('2.0'))
+
+    def test_null_with_precision(self):
+        IntegerModel.objects.create()
+        obj = IntegerModel.objects.annotate(
+            null_round=Round('normal', 5),
+        ).first()
+        self.assertIsNone(obj.null_round)
+
+    def test_decimal_with_precision(self):
+        DecimalModel.objects.create(n1=-5.75, n2=Pi())
+        obj = DecimalModel.objects.annotate(
+            n1_round=Round('n1', 1),
+            n2_round=Round('n2', 5),
+        ).first()
+        self.assertIsInstance(obj.n1_round, Decimal)
+        self.assertIsInstance(obj.n2_round, Decimal)
+        self.assertAlmostEqual(obj.n1_round, obj.n1, places=1)
+        self.assertAlmostEqual(obj.n2_round, obj.n2, places=5)
+
+    def test_float_with_precision(self):
+        FloatModel.objects.create(f1=-5.75, f2=Pi())
+        obj = FloatModel.objects.annotate(
+            f1_round=Round('f1', 1),
+            f2_round=Round('f2', 5),
+        ).first()
+        self.assertIsInstance(obj.f1_round, float)
+        self.assertIsInstance(obj.f2_round, float)
+        self.assertAlmostEqual(obj.f1_round, obj.f1, places=1)
+        self.assertAlmostEqual(obj.f2_round, obj.f2, places=5)
+
+    def test_integer_with_precision(self):
+        IntegerModel.objects.create(small=-5, normal=3)
+        obj = IntegerModel.objects.annotate(
+            small_round=Round('small', 1),
+            normal_round=Round('normal', 5),
+        ).first()
+        self.assertIsInstance(obj.small_round, float)
+        self.assertIsInstance(obj.normal_round, float)
+        self.assertAlmostEqual(obj.small_round, obj.small, places=1)
+        self.assertAlmostEqual(obj.normal_round, obj.normal, places=5)
+
+    @unittest.skipIf(connection.vendor == 'sqlite', 'Negative precision not supported by SQLite.')
+    def test_null_with_negative_precision(self):
+        IntegerModel.objects.create()
+        obj = IntegerModel.objects.annotate(
+            null_round=Round('normal', -1),
+        ).first()
+        self.assertIsNone(obj.null_round)
+
+    @unittest.skipIf(connection.vendor == 'sqlite', 'Negative precision not supported by SQLite.')
+    def test_decimal_with_negative_precision(self):
+        DecimalModel.objects.create(n1=365.25)
+        obj = DecimalModel.objects.annotate(n1_round=Round('n1', -1)).first()
+        self.assertIsInstance(obj.n1_round, Decimal)
+        self.assertEqual(obj.n1_round, 370)
+
+    @unittest.skipIf(connection.vendor == 'sqlite', 'Negative precision not supported by SQLite.')
+    def test_float_with_negative_precision(self):
+        FloatModel.objects.create(f1=365.25)
+        obj = FloatModel.objects.annotate(f1_round=Round('f1', -1)).first()
+        self.assertIsInstance(obj.f1_round, float)
+        self.assertEqual(obj.f1_round, 370)
+
+    @unittest.skipIf(connection.vendor == 'sqlite', 'Negative precision not supported by SQLite.')
```

Comment:
We should add a new database feature or add these tests to the `DatabaseFeatures.django_test_skips` on SQLite.

Reply:
Good point. Added to `DatabaseFeatures.django_test_skips`.